include/cupla/defines.hpp

```diff
@@ -111,3 +111,17 @@
 #ifndef CUPLA_HEADER_ONLY_FUNC_SPEC
 #   define CUPLA_HEADER_ONLY_FUNC_SPEC
 #endif
+
+/*! device compiler path
+ *
+ * Enabled if the compiler processes currently a separate compile path for the device code
+ *
+ * @attention value is always 0 for alpaka CPU accelerators
+ *
+ * Value is 1 if device path is compiled else 0
+ */
+#if defined(__CUDA_ARCH__) || ( defined(__HIP_DEVICE_COMPILE__) && __HIP_DEVICE_COMPILE__== 1 && defined(__HIP__) )
+    #define CUPLA_DEVICE_PATH 1
+#else
+    #define CUPLA_DEVICE_PATH 0
+#endif
```

Comment:
I think the name `CUPLA_DEVICE_PATH` may be confusing, due to both alpaka calling everything devices, and path could be interpreted as a filesystem path. When used elsewhere out of context may be surprising. Perhaps a more verbose name, along the lines of `CUPLA_SEPARATE_DEVICE_COMPILE_PATH` or `CUPLA_SEPARATE_DEVICE_COMPILATION` would be more fitting? That said, the comment explains it all well, so even as is it's not bad.

Reply:
You are right, I will take the HIP version `__HIP_DEVICE_COMPILE__` as base an switch to `CUPLA_DEVICE_COMPILE`.
I will avoid `SEPARATE` to avoid confusion with the feature separate compile for HIP and CUDA.

Reply:
Yes, good point about separate, I did not consider this.